hudi-common/src/main/java/org/apache/hudi/common/util/MarkerUtils.java

```diff
@@ -216,4 +216,4 @@ public static void deleteMarkerTypeFile(FileSystem fileSystem, String markerDir)
       throw new HoodieIOException(ioe.getMessage(), ioe);
     }
   }
-}
+}
```

Comment:
Comment about L186. 
I see we filter for entire file path when filtering for MARKERS.*. 
shouldn't we fetch just filename and then do filtering? what incase basepath has "MARKERS" string in them? 

I fixed this in TwoToOneDowngrade recreation when we delete the timeline based markers. 
```
Predicate<FileStatus> prefixFilter = pathStr -> pathStr.getPath().getName().contains(MARKERS_FILENAME_PREFIX);
    List<String> markerDirSubPaths = Arrays.stream(fileStatuses)
        .filter(prefixFilter)
        .map(fileStatus -> fileStatus.getPath().toString())
        .collect(Collectors.toList());
```

Reply:
pathStr.getPath().getName().startWith(MARKERS_FILENAME_PREFIX) seems like the right way to me. is there a place where we should change this

Reply:
Yes.  I made a pass of all filters we use and properly adjusted them.